overlays/custom-packages/labwc/labwc-colored-borders.patch

```diff
@@ -0,0 +1,521 @@
+diff --git a/.gitignore b/.gitignore
+new file mode 100644
+index 0000000..02f1769
+--- /dev/null
++++ b/.gitignore
+@@ -0,0 +1,2 @@
++result
++.vscode
+diff --git a/include/ssd-internal.h b/include/ssd-internal.h
+index 9fe0ebf..6dfc2d4 100644
+--- a/include/ssd-internal.h
++++ b/include/ssd-internal.h
+@@ -118,8 +118,8 @@ struct ssd_part *add_scene_button(
+ 	struct wlr_buffer *icon_buffer, int x, struct view *view);
+ struct ssd_part *add_scene_button_corner(
+ 	struct wl_list *part_list, enum ssd_part_type type,
+-	enum ssd_part_type corner_type, struct wlr_scene_tree *parent,
+-	struct wlr_buffer *corner_buffer, struct wlr_buffer *icon_buffer,
++	enum ssd_part_type corner_type, struct wlr_scene_tree *parent, 
++	float *bg_color, struct wlr_buffer *corner_buffer, struct wlr_buffer *icon_buffer,
```

Comment:
The common practice to extend (or hack) with additional parameters is to add them to the end of the parameter list - like to add `float *bg_color` in the end, not in the middle. The same applies on extending enums, structs.
If you have a specific reason to add it randomly in the middle, please leave a code comment.

Reply:
The function 
```
struct ssd_part *
add_scene_button(struct wl_list *part_list, enum ssd_part_type type,
		struct wlr_scene_tree *parent, float *bg_color,
		struct wlr_buffer *icon_buffer, struct wlr_buffer *hover_buffer,
		int x, struct view *view)
```
has _bg_color_  parameter and is used in the 

```
struct ssd_part *
add_scene_button_corner(struct wl_list *part_list, enum ssd_part_type type,
		enum ssd_part_type corner_type, struct wlr_scene_tree *parent,
		struct wlr_buffer *corner_buffer, struct wlr_buffer *icon_buffer,
		struct wlr_buffer *hover_buffer, int x, struct view *view)
```
I just tried to unify the parameters ordering. It helps avoiding mistakes.

Reply:
Thanks for clarifying your thinking here. If I'm not mistaken, the parameter ordering comes from the upstream project. How could we check if this contribution would be accepted to the upstream project for easier maintenance?

Reply:
I asked the maintainers and waiting for an answer. This solution doesn't change the overall logic (but must do it) and just substitutes the colors in a few code snippets. To introduce the colored window borders as a feature the theme handling must be revised (in my opinion).